02_bubble/nicholsk/main.go

```diff
@@ -0,0 +1,45 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	fmt.Fprintln(out, bubble([]int{3, 2, 1, 5}))
+}
+
+func bubble(s []int) []int {
+
+	// track whether the sort is complete
+	sorted := false
+
+	for !sorted {
+
+		// track whether any swaps took place in current iteration
+		swaps := false
+
+		// iterate every index in the list, starting at index 1
+		for i := 1; i < len(s); i++ {
+
+			// compare element to previous
+			if s[i-1] > s[i] {
+
+				// if less than previous then swap elements
+				s[i], s[i-1] = s[i-1], s[i]
```

Comment:
you haven't created a copy of the input slice and you are there for modifying the input.
create a test case that fails when checking that input is left unmodified and fix with `copy()`